pygmt/helpers/tempfile.py

```diff
@@ -104,3 +105,54 @@ def loadtxt(self, **kwargs):
             Data read from the text file.
         """
         return np.loadtxt(self.name, **kwargs)
+
+
+@contextmanager
+def tempfile_from_geojson(geojson):
+    """
+    Saves any geo-like Python object which implements ``__geo_interface__``
+    (e.g. a geopandas GeoDataFrame) to a temporary OGR_GMT text file.
+
+    Parameters
+    ----------
+    geojson : geopandas.GeoDataFrame
+        A geopandas GeoDataFrame, or any geo-like Python object which
+        implements __geo_interface__, i.e. a GeoJSON
+
+    Yields
+    ------
+    tmpfilename : str
+        A temporary OGR_GMT format file holding the geographical data.
+        E.g. 'track-1a2b3c4.tsv'.
+    """
+    with GMTTempFile(suffix=".gmt") as tmpfile:
+        os.remove(tmpfile.name)  # ensure file is deleted first
+        ogrgmt_kwargs = dict(filename=tmpfile.name, driver="OGR_GMT", mode="w")
+        try:
+            # Using geopandas.to_file to directly export to OGR_GMT format
+            geojson.to_file(**ogrgmt_kwargs)
+        except AttributeError:
+            # pylint: disable=import-outside-toplevel
+            # Other 'geo' formats which implement __geo_interface__
+            import json
+
+            import fiona
+            import geopandas as gpd
+
+            with fiona.Env():
+                jsontext = json.dumps(geojson.__geo_interface__)
+                # Do Input/Output via Fiona virtual memory
+                with fiona.io.MemoryFile(file_or_bytes=jsontext.encode()) as memfile:
+                    geoseries = gpd.GeoSeries.from_file(filename=memfile)
+                    geoseries.to_file(**ogrgmt_kwargs)
+
+                #     with memfile.open(driver="GeoJSON") as collection:
+                #         # Get schema from GeoJSON
+                #         schema = collection.schema
+                # # Write to temporary OGR_GMT format file
+                # with fiona.open(
+                #    fp=tmpfile.name, mode="w", driver="OGR_GMT", schema=schema
+                # ) as ogrgmtfile:
+                #     ogrgmtfile.write(geojson.__geo_interface__)
```

Comment:
This was my failed attempt to convert a GeoJSON string to an OGR_GMT (*.gmt) format file purely using `fiona` (i.e. no `geopandas` installed). The problem I found was that a schema had to be set, and that requires a lot of lines of code to do, which is why I stuck to using `geopandas` in the lines above.

If someone can figure out a good way to solve the schema problem, that would be fantastic!

Reply:
Ok, I've removed this chunk of unusable code in 34de789e30e1c2b71fff11840baeb0e504bdd734. Hopefully this can be revisited in the future so that GeoJSON objects can be converted into OGR_GMT purely using `fiona` without the need for installing `geopandas` (a bit of a big dependency).